### DISCUSSION TITLE
Unable to update user details. Update API giving INTERNAL SERVER ERROR

### DISCUSSION THREAD

**Discussion by paritshivani** [Sunbird-Ed/Community#422] [Bugs] [answered]
Hello folks! Hope you are doing well!
Please help in following issue
We are trying to update user preferences and getting error to update user api
**url:** /learner/user/v2/update
**CURL:**

curl '-------/learner/user/v2/update' \
  -X 'PATCH' \
  -H 'Connection: keep-alive' \
  -H 'sec-ch-ua: " Not A;Brand";v="99", "Chromium";v="98", "Google Chrome";v="98"' \
  -H 'ts: 2023-03-03T13:18:10+05:30' \
  -H 'X-msgid: c4f2c8a8-6755-932c-4ed9-e34b4d7a34ee' \
  -H 'X-Org-code: 0137299712231669762' \
  -H 'X-Source: web' \
  -H 'X-User-ID: 0ffac1cc-6f30-41ec-9911-5171ff2d911a' \
  -H 'sec-ch-ua-platform: "Linux"' \
  -H 'X-Channel-Id: 0137299712231669762' \
  -H 'X-App-Id: dev.dev.portal' \
  -H 'X-App-Version: 4.4.1' \
  -H 'X-Request-ID: c4f2c8a8-6755-932c-4ed9-e34b4d7a34ee' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.102 Safari/537.36' \
  -H 'Content-Type: application/json' \
  -H 'Accept: application/json' \
  -H 'X-Session-ID: ----------------' \
  -H 'X-Device-ID: c746e183f7bb3c4731ac02de4fbf7f61' \
  -H 'Origin:----------------' \
  -H 'Sec-Fetch-Site: same-origin' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Referer: --------------/profile' \
  -H 'Accept-Language: en-GB,en-US;q=0.9,en;q=0.8' \
  -H 'Cookie: -------------------' \
  --data-raw '{"params":{},"request":{"framework":{"board":["Environment and Climate"],"medium":["Citizen Engagement"],"gradeLevel":["Solutions Hackathon"],"id":"nulplearnathon"},"userId":"0ffac1cc-6f30-41ec-9911-5171ff2d911a"}}' \
  --compressed`

**Response:**

```
{
    "id": "api.user.update",
    "ver": "v2",
    "ts": "2023-03-01 12:19:14:415+0000",
    "params": {
        "resmsgid": null,
        "msgid": "0327e172-cb36-f3e8-2175-fdcc462de79a",
        "err": "INTERNAL_ERROR",
        "status": "INTERNAL_ERROR",
        "errmsg": "Process failed,please try again later."
    },
    "responseCode": "SERVER_ERROR",
    "result": {}
}
```

**Learner service Logs:** 

```
{\"err\":\"500\",\"stacktrace\":\"controllers.BaseController.createCommonExceptionResponse(Base
Controller.java:530)\\
ncontrollers.BaseCo\",\"errtype\":\"INTERNAL_ERROR\",\"requestid\":\"fe2a2ae1-8148-8fda-
84bc-7aa53764c1e0\"}}","lname":"TelemetryEventLogger","tname":"Thread-
18769","level":"INFO","HOSTNAME":"learner-565b4c5cd9-th9mp","application.home":"/home/
sunbird/learner/learning-service-1.0-SNAPSHOT"}
{"timestamp":"2023-03-
01T10:40:52.886Z","msg":"{\"eid\":\"ERROR\",\"ets\":1677667252886,\"ver\":\"3.0\",\"mid\":\"fe2
a2ae1-8148-8fda-84bc-7aa53764c1e0\",\"actor\":{\"id\":\"0ffac1cc-6f30-41ec-9911-
5171ff2d911a\",\"type\":\"User\"},\"context\":{\"channel\":\"0137299712231669762\",\"pdata\":
{\"id\":\"local.sunbird.portal\"},\"env\":\"User\",\"did\":\"eade363e751518891a3bb698534c2a12\",\
"cdata\":[{\"id\":\"fe2a2ae1-8148-8fda-84bc-7aa53764c1e0\",\"type\":\"Request\"}],\"rollup\":
{}},\"edata\":{\"err\":\"INTERNAL_ERROR\",\"stacktrace\":\"Process failed,please try again later. 
controllers.BaseController.createCommonExceptionResponse(BaseController.java:530)controllers.
BaseController.lambda$actorResponseHandler$0(BaseController.java:673)java.base/
java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:642)java.base/
java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:478)java.base/
java.lang.Thread.run(Thread.java:829)\",\"errtype\":\"api_access\",\"requestid\":\"fe2a2ae1-8148-
8fda-84bc-7aa53764c1e0\"}}","lname":"TelemetryEventLogger","tname":"Thread-
18769","level":"INFO","HOSTNAME":"learner-565b4c5cd9-th9mp","application.home":"/home/
sunbird/learner/learning-service-1.0-SNAPSHOT"}
{"timestamp":"2023-03-
01T10:40:52.886Z","msg":"{\"eid\":\"LOG\",\"ets\":1677667252886,\"ver\":\"3.0\",\"mid\":\"fe2a2
ae1-8148-8fda-84bc-7aa53764c1e0\",\"actor\":{\"id\":\"0ffac1cc-6f30-41ec-9911-
5171ff2d911a\",\"type\":\"User\"},\"context\":{\"channel\":\"0137299712231669762\",\"pdata\":
{\"id\":\"local.sunbird.portal\"},\"env\":\"User\",\"did\":\"eade363e751518891a3bb698534c2a12\",\
"cdata\":[{\"id\":\"fe2a2ae1-8148-8fda-84bc-7aa53764c1e0\",\"type\":\"Request\"}],\"rollup\":
{}},\"edata\":{\"level\":\"info\",\"type\":\"Api_access\",\"message\":\"{eid='LOG', 
edata={level=trace, requestid=fe2a2ae1-8148-8fda-84bc-7aa53764c1e0, type=system, 
message=EXIT LOG: method : PATCH, url: /v2/user/update , For Operation : updateUserV2, 
params=[{msgid=fe2a2ae1-8148-8fda-84bc-7aa53764c1e0, errmsg=Process failed,please try again 
later., resmsgid=null, err=INTERNAL_ERROR, status=FAILED, 
responseCode=500}]}}\",\"params\":[]}}","lname":"TelemetryEventLogger","tname":"Thread-
18769","level":"INFO","HOSTNAME":"learner-565b4c5cd9-th9mp","application.home":"/home/
sunbird/learner/learning-service-1.0-SNAPSHOT"
```

@gandham-santhosh , @saiakhil46 , @coolbung

**Comment by Krishnaj20** [comment]
@reshmi-nair - can you check this thread and respond please?

@chitranshu-keshav

**Reply by reshmi-nair** [reply]
@paritshivani Pls check the value configured in sunbird.system_settings table in cassandra for "userProfileConfig" property. Framework feilds are validate against the configuration in the property. Also make sure the "/v1/framework/read" API is configured properly to validate the framework against content service.

**Reply by Krishnaj20** [reply]
@paritshivani - since we haven't received any response, we are assuming that your query is resolved. Hence marking it as answered.

**Comment by Krishnaj20** [comment, marked as answer]
@paritshivani Pls check the value configured in sunbird.system_settings table in cassandra for "userProfileConfig" property. Framework feilds are validate against the configuration in the property. Also make sure the "/v1/framework/read" API is configured properly to validate the framework against content service.
